src/components/Header/Header.test.js

```diff
@@ -0,0 +1,54 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Header } from "./Header";
+
+describe("<Header />", () => {
+  describe("should render all buttons", () => {
+    it("renders the List of Puns button", () => {
+      render(<Header />)
+      const listButton = screen.getByRole("button", { name: /list of puns/i });
+
+      expect(listButton).toBeInTheDocument()
+    })
+    it("renders the Add Pun button", () => {
+      render(<Header />)
+      const addButton = screen.getByRole("button", { name: /add pun/i });
+
+      expect(addButton).toBeInTheDocument()
+    })
+    it("renders the Ranking button", () => {
+      render(<Header />)
+      const rankingButton = screen.getByRole('button', {name: /ranking/i})
+
+      expect(rankingButton).toBeInTheDocument()
+    })
+  })
+  describe("when the List of Puns button is clicked", () => {
+    it("changes states to render list of Puns", () => {
```

Comment:
A descrição poderia estar mais condizente com o componente se fosse algo do tipo: "calls the setListState and setAddState functions"